src/librustc_back/target/mipsel_unknown_linux_musl.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright 2014 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use target::Target;
+
+pub fn target() -> Target {
+    Target {
+        llvm_target: "mipsel-unknown-linux-musl".to_string(),
+        target_endian: "little".to_string(),
+        target_pointer_width: "32".to_string(),
+        arch: "mips".to_string(),
```

Comment:
Oh interesting! That may be a bug in the target triple, however, depending on how you look at it. Either that or we may just want to make sure that we're consistent. Some data points:

* The newly-added `powerpc64le-unknown-linux-gnu` triple has `powerpc64le` as the `target_arch`.
* There's some `cfg(target_arch = "mipsel")` directives in the standard library, which apparently are doing nothing.
* We don't necessarily require the arch in the triple to equal the arch in `target_arch`, for example `armv7` recently added has the `target_arch` of `arm`.

I think that I'd prefer to lean towards leaving this as `mips` and tweaking the `powerpc64le` triple to have `powerpc64` as the `target_arch`. That seems the most consistent to me and also the most ergonomic. The two triples can still be differentiated by looking at `target_endian`